### ISSUE TITLE
bcftools mpileup cannot parse header line (containing pipe characters)

### ISSUE THREAD

**Issue by blue-moon22** [samtools/bcftools#1032]
Hi there,

I'm running command bcftools mpileup -f $refCatalog $bamFile and getting this error:
[E::bcf_hdr_parse_line] Could not parse the header line: "##contig=<ID=gi|284919779|emb|FN554766.1|:931349-931645,931649-932062,length=711>"

Saw an issue was closed a while ago on headers containing commas. This seems to be only happening with headers containing pipes, so could be similar problem with pipe characters.

Is there a quick fix for this? Could I replace the pipes in the headers of refCatalog and bamFile?

**Comment by pd3** [comment]
Yes, the quick fix is to rename to a more sensible contig names. Any potential changes in the htslib parsing code will take long.

**Comment by jmarshall** [comment]
The error message is because what it thinks is a field name (`931649-932062`) isn't followed by `=`. So you need to put quotes around your contig ID that contains a comma.
